### DISCUSSION TITLE
Solving first order hyperbolic equations with py-pde

### DISCUSSION THREAD

**Discussion by ocichorek** [zwicker-group/py-pde#270] [Q&A] [answered]
First of all, thanks for sharing this awesome package.

Is py-pde is capable of solving first order hyperbolic equations, e.g. transport equations?

![obraz](https://user-images.githubusercontent.com/98961607/184921096-6cdb84ab-7ac8-403b-936f-d17dc25ad157.png)

From the mathematical point of view, for a constant value of c the equation needs just one boundary condition (left bc for positive values of c; right bc for negative values of c). But according to the reference manual each axis needs to have  boundary conditions specified on its both ends. I was wondering whether there is a workaround, which would allow to specify only one boundary condition for the x-axis?

Ultimately, I would like to solve a set of two coupled continuity equations:

![obraz](https://user-images.githubusercontent.com/98961607/184920062-fa919f5f-39dd-47c5-b5ce-50f51de3a4c4.png)

where v_a, v_i and beta are constant coefficients.

**Comment by david-zwicker** [comment, marked as answer]
I'm afraid that this is not currently possible. You could try your luck with setting an arbitrary condition on the right, but I know that advective terms, like the second ones on the right, are difficult to treat correctly and I haven't explored this in any way.
